beacon-chain/operations/synccommittee/contribution.go

```diff
@@ -21,17 +21,23 @@ func (s *Store) SaveSyncCommitteeContribution(cont *prysmv2.SyncCommitteeContrib
 		return nilContributionErr
 	}
 
-	contributions, err := s.SyncCommitteeContributions(cont.Slot)
+	s.contributionLock.Lock()
+	defer s.contributionLock.Unlock()
+
+	item, err := s.contributionCache.PopByKey(syncCommitteeKey(cont.Slot))
 	if err != nil {
 		return err
 	}
 
-	s.contributionLock.Lock()
-	defer s.contributionLock.Unlock()
 	copied := copyutil.CopySyncCommitteeContribution(cont)
 
 	// Contributions exist in the queue. Append instead of insert new.
-	if contributions != nil {
+	if item != nil {
```

Comment:
in the event the slot exists in the cache, pruning is skipped in this conditional path. Its only done if a new contribution slot is saved further below

Reply:
addressed offline, slot is the key. We only care to prune when there's new slot that gets inserted